File: numberguess.cpp
#include <iostream>
#include <cstdlib>
#include <cctype>

using namespace std;

class NumberGuess {
	private:
		int answer;
		int numGuesses;
	public:
		NumberGuess() {
			answer = rand()%100;
			numGuesses = 0;
		}

		int Guess(int guess) {
			numGuesses += 1;
			if (guess < answer) {
				return -1;
			} else if (guess > answer) {
				return 1;
			} else {
				return 0;
			}
		}
		int Guesses() {
			return numGuesses;
		}
};



int main()
{
	string playAgain = "Y";
	int guess;
	int ngr;

	while (tolower(playAgain[0]) == 'y') {
		NumberGuess ng;
		do {
			cout << "Please guess a number: ";
			cin >> guess;
			ngr = ng.Guess(guess);
			if (ngr < 0) {
				cout << guess << " is too low!" << endl;
			} else if (ngr > 0) {
				cout << guess << " is too high!!" << endl;
			}
		} while (ngr != 0);
		cout << "Guessed it after " << ng.Guesses() << " guesses!" << endl;
		cout << "Play again? ";
		cin >> playAgain;
	}
	return 0;
}

File: numberGuess.scala
import scala.io.StdIn._

//> using scala 3.7.1

@main
def numberGuess(): Unit =

        val rand = new scala.util.Random
        var playAgain = "yes"


        while (playAgain.toLowerCase().startsWith("y")) {
                var answer = rand.nextInt(100)+1
                var guess = 0
                var guesses = 0

                while (guess != answer) {
                        var guessStr = readLine("Guess :>")
                        guess = guessStr.toIntOption.getOrElse(-1)
                        guesses += 1

                        if (guess < 0) {
                                println(s"Please inpoout a number!!")
                        } else if (guess < answer) {
                                println(s"$guess is too low!")
                        } else if (guess > answer) {
                                println(s"$guess is too high!")
                        }
                }
                println(s"You got it in $guesses guesses!")
                playAgain = readLine("Play again [y/n] :>")
        }
        println(s"Thank you for playing!")






